### PR TITLE
Add neutrally stratified Ekman layer experiment and anelastic driver

### PR DESCRIPTION
**Pull request by ilopezgp**
### Description

This PR adds a new neutrally stratified Ekman layer experiment to the LES/SCM library, and an anelastic driver to run it. The anelastic driver can be switched from DG to FV with a single boolean.

This configuration removes the effect of density stratification from the dynamics (in theory...), helping debug the source code in the DG and FV configurations.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by ilopezgp** [comment]
bors try

File: experiments/AtmosLES/ekman_layer_model.jl [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+using ClimateMachine.GenericCallbacks
+using ClimateMachine.Mesh.Filters
+using ClimateMachine.Mesh.Grids
+using ClimateMachine.ODESolvers
+using ClimateMachine.Thermodynamics
+using ClimateMachine.TurbulenceClosures
+using ClimateMachine.TurbulenceConvection
+using ClimateMachine.VariableTemplates
+using ClimateMachine.BalanceLaws
+import ClimateMachine.BalanceLaws: source
```
Comment (charleskawczynski) [inline experiments/AtmosLES/ekman_layer_model.jl:39]:
```suggestion
import ClimateMachine.BalanceLaws: source, prognostic_vars
```

File: experiments/AtmosLES/ekman_layer_model.jl [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    f_coriolis::FT
+    "Eastward geostrophic velocity `[m/s]` (Base)"
+    u_geostrophic::FT
+    "Eastward geostrophic velocity `[m/s]` (Slope)"
+    u_slope::FT
+    "Northward geostrophic velocity `[m/s]`"
+    v_geostrophic::FT
+end
+NeutralBLGeostrophic(::Type{FT}, args...) where {FT} =
+    NeutralBLGeostrophic{Momentum, FT}(args...)
```
Comment (charleskawczynski) [inline experiments/AtmosLES/ekman_layer_model.jl:64]:
```suggestion
prognostic_vars(::NeutralBLGeostrophic) = (Momentum(),)
```

File: experiments/AtmosLES/ekman_layer_model.jl [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+import CLIMAParameters
+
+using ClimateMachine.Atmos: altitude, recover_thermo_state, density
+
+"""
+  NeutralBL Geostrophic Forcing (Source)
+"""
+struct NeutralBLGeostrophic{PV <: Momentum, FT} <: TendencyDef{Source, PV}
```
Comment (charleskawczynski) [inline experiments/AtmosLES/ekman_layer_model.jl:53]:
```suggestion
struct NeutralBLGeostrophic{FT} <: TendencyDef{Source}
```

File: experiments/AtmosLES/ekman_layer_model.jl [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+    u_geostrophic::FT
+    "Eastward geostrophic velocity `[m/s]` (Slope)"
+    u_slope::FT
+    "Northward geostrophic velocity `[m/s]`"
+    v_geostrophic::FT
+end
+NeutralBLGeostrophic(::Type{FT}, args...) where {FT} =
+    NeutralBLGeostrophic{Momentum, FT}(args...)
+
+function source(s::NeutralBLGeostrophic{Momentum}, m, args)
```
Comment (charleskawczynski) [inline experiments/AtmosLES/ekman_layer_model.jl:66]:
```suggestion
function source(::Momentum, s::NeutralBLGeostrophic, m, args)
```

File: experiments/AtmosLES/ekman_layer_model.jl [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+    ẑ = vertical_unit_vector(m, aux)
+    fkvector = f_coriolis * ẑ
+    # Accumulate sources
+    return -fkvector × (state.ρu .- state.ρ * u_geo)
+end
+
+"""
+  NeutralBL Sponge (Source)
+"""
+struct NeutralBLSponge{PV <: Momentum, FT} <: TendencyDef{Source, PV}
```
Comment (charleskawczynski) [inline experiments/AtmosLES/ekman_layer_model.jl:82]:
```suggestion
struct NeutralBLSponge{FT} <: TendencyDef{Source}
```

File: experiments/AtmosLES/ekman_layer_model.jl [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+    "Eastward geostrophic velocity `[m/s]` (Base)"
+    u_geostrophic::FT
+    "Eastward geostrophic velocity `[m/s]` (Slope)"
+    u_slope::FT
+    "Northward geostrophic velocity `[m/s]`"
+    v_geostrophic::FT
+end
+
+NeutralBLSponge(::Type{FT}, args...) where {FT} =
+    NeutralBLSponge{Momentum, FT}(args...)
```
Comment (charleskawczynski) [inline experiments/AtmosLES/ekman_layer_model.jl:100]:
```suggestion
prognostic_vars(::NeutralBLSponge) = (Momentum(), )
```

File: experiments/AtmosLES/ekman_layer_model.jl [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+
+function source(s::NeutralBLSponge{Momentum}, m, args)
```
Comment (charleskawczynski) [inline experiments/AtmosLES/ekman_layer_model.jl:102]:
```suggestion
function source(::Momentum, s::NeutralBLSponge, m, args)
```

File: experiments/AtmosLES/ekman_layer_model.jl [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+
+    q_sfc = FT(0)
+    θ_sfc = T_surf_ref(param_set)
+    g = compressibility isa Compressible ? (Gravity(),) : ()
+
+    # Assemble source components
+    source_default = (
+        g...,
+        NeutralBLSponge(
+            FT,
```
Comment (charleskawczynski) [inline experiments/AtmosLES/ekman_layer_model.jl:199]:
```suggestion
        NeutralBLSponge{FT}(
```

**Review by charleskawczynski — COMMENTED** [review]
Sorry for not reviewing until now, I wanted to let #2109 go through first. The suggested changes should fix the pieces that #2109 broke in this PR.

File: experiments/AtmosLES/ekman_layer_model.jl [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+            zmax,
+            z_sponge,
+            α_max,
+            γ,
+            u_geostrophic,
+            u_slope,
+            v_geostrophic,
+        ),
+        NeutralBLGeostrophic(
+            FT,
```
Comment (charleskawczynski) [inline experiments/AtmosLES/ekman_layer_model.jl:207]:
```suggestion
        NeutralBLGeostrophic{FT}(
```

File: experiments/AtmosLES/ekman_layer_model.jl [hunk 10]
```diff
@@ -0,0 +1,301 @@
...
+
+    q_sfc = FT(0)
+    θ_sfc = T_surf_ref(param_set)
+    g = compressibility isa Compressible ? (Gravity(),) : ()
+
+    # Assemble source components
+    source_default = (
+        g...,
+        NeutralBLSponge{FT}(
+            FT,
```
Comment (charleskawczynski) [inline experiments/AtmosLES/ekman_layer_model.jl:197]:
```suggestion
        NeutralBLSponge{FT}(
```

Comment (ilopezgp) [reply experiments/AtmosLES/ekman_layer_model.jl:197]: Yes, I just saw these. Fixing it now.

**Comment by ilopezgp** [comment]
It should be good now, I rebased and it runs

File: test/Atmos/EDMF/ekman_layer_anelastic1d.jl [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+            zmax,
+            param_set,
+            model;
+            hmax = FT(40),
+            solver_type = ode_solver_type,
+            fv_reconstruction = FVLinear(),
+        )
+    else
+        # DG polynomial order
+        N = 1
```
Comment (charleskawczynski) [inline test/Atmos/EDMF/ekman_layer_anelastic1d.jl:91]: Can we somehow write this more compactly?

**Review by charleskawczynski — COMMENTED** [review]
Can we add the test to buildkite?

**Comment by ilopezgp** [comment]
Agreed. I am also going to make the experiment file more flexible to enable calling EDMF (@yairchn and I are now using this setup). I will set as the default the DG, single stack, anelastic1D configuration with constant kinematic viscosity and no EDMF, and that one will be tested.

File: experiments/AtmosLES/ekman_layer_model.jl [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+        moisture_bc = PrescribedMoistureFlux((state, aux, t) -> FT(0))
+    else
+        @warn @sprintf(
+            """
+%s: unrecognized surface flux; using 'prescribed'""",
+            surface_flux,
+        )
+    end
+
+    moisture_bcs = ()
```
Comment (yairchn) [inline experiments/AtmosLES/ekman_layer_model.jl:247]: why is this here if 'moisture_bc' is defined above ?

File: test/Atmos/EDMF/ekman_layer.jl [hunk 13]
```diff
@@ -0,0 +1,271 @@
...
+        nelem_vert = 80
+        ref_state = HydrostaticState(
+            DryAdiabaticProfile{FT}(param_set),
+            ;
+            subtract_off = false,
+        )
+        output_prefix = string("EL_", str_comp, "_FVM")
+        fv_reconstruction = FVLinear()
+    else
+        N = 4
```
Comment (yairchn) [inline test/Atmos/EDMF/ekman_layer.jl:129]: do you want to change to 
N=1
nelem_vert = 80
here?

File: test/Atmos/EDMF/ekman_layer.jl [hunk 14]
```diff
@@ -0,0 +1,271 @@
...
+        driver_config.grid,
+        solver_config.dg.state_auxiliary,
+        varsindex(vsa, :turbconv),
+    )
+    # ---
+
+    dgn_config = config_diagnostics(driver_config)
+
+    # boyd vandeven filter
+    cb_boyd = GenericCallbacks.EveryXSimulationSteps(1) do
```
Comment (yairchn) [inline test/Atmos/EDMF/ekman_layer.jl:190]: is this used?

Comment (ilopezgp) [reply test/Atmos/EDMF/ekman_layer.jl:190]: No, but I want to leave it there so we can easily plug it in the callback. This is also useful for issue #2116

File: test/Atmos/EDMF/ekman_layer.jl (hunk 13 above)
Comment (ilopezgp) [reply test/Atmos/EDMF/ekman_layer.jl:129]: not at the moment, I want to leave the default for #2116

File: experiments/AtmosLES/ekman_layer_model.jl (hunk 12 above)
Comment (ilopezgp) [reply experiments/AtmosLES/ekman_layer_model.jl:247]: good catch

**Comment by ilopezgp** [comment]
bors try

**Comment by ilopezgp** [comment]
This is ready @yairchn @charleskawczynski

**Review by yairchn — APPROVED** [review]
